src/main/java/com/conveyal/datatools/manager/models/FeedSourceSummary.java

```diff
@@ -0,0 +1,548 @@
+package com.conveyal.datatools.manager.models;
+
+import com.conveyal.datatools.editor.utils.JacksonSerializers;
+import com.conveyal.datatools.manager.persistence.Persistence;
+import com.conveyal.gtfs.validator.ValidationResult;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.annotation.JsonSerialize;
+import com.google.common.collect.Lists;
+import com.mongodb.client.model.Accumulators;
+import com.mongodb.client.model.Projections;
+import com.mongodb.client.model.Sorts;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static com.mongodb.client.model.Aggregates.group;
+import static com.mongodb.client.model.Aggregates.limit;
+import static com.mongodb.client.model.Aggregates.lookup;
+import static com.mongodb.client.model.Aggregates.match;
+import static com.mongodb.client.model.Aggregates.project;
+import static com.mongodb.client.model.Aggregates.replaceRoot;
+import static com.mongodb.client.model.Aggregates.sort;
+import static com.mongodb.client.model.Aggregates.unwind;
+import static com.mongodb.client.model.Filters.in;
+
+public class FeedSourceSummary {
+    private static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("yyyyMMdd");
+
+    public String projectId;
+
+    public String id;
+
+    public String name;
+    public boolean deployable;
+    public boolean isPublic;
+
+    @JsonSerialize(using = JacksonSerializers.LocalDateIsoSerializer.class)
+    @JsonDeserialize(using = JacksonSerializers.LocalDateIsoDeserializer.class)
+    public LocalDate lastUpdated;
+
+    public List<String> labelIds = new ArrayList<>();
+
+    public String deployedFeedVersionId;
+
+    @JsonSerialize(using = JacksonSerializers.LocalDateIsoSerializer.class)
+    @JsonDeserialize(using = JacksonSerializers.LocalDateIsoDeserializer.class)
+    public LocalDate deployedFeedVersionStartDate;
+
+    @JsonSerialize(using = JacksonSerializers.LocalDateIsoSerializer.class)
+    @JsonDeserialize(using = JacksonSerializers.LocalDateIsoDeserializer.class)
+    public LocalDate deployedFeedVersionEndDate;
+
+    public Integer deployedFeedVersionIssues;
+
+    public LatestValidationResult latestValidation;
+
+    public FeedSourceSummary() {
+    }
+
+    public FeedSourceSummary(String projectId, Document feedSourceDocument) {
+        this.projectId = projectId;
+        this.id = feedSourceDocument.getString("_id");
+        this.name = feedSourceDocument.getString("name");
+        this.deployable = feedSourceDocument.getBoolean("deployable");
+        this.isPublic = feedSourceDocument.getBoolean("isPublic");
+        this.labelIds = feedSourceDocument.getList("labelIds", String.class);
```

Comment:
The UI currently assumes that `labelIds` is non-null and at least an empty list. I think this line may cause `null` to be returned in the response and break the UI.
@philip-cline What do you think?

Reply:
I haven't experienced that in a project without labelIds, it seems to return an empty array

Reply:
I've updated to always return an empty array if labelIds is null.